src/app-layer-htp.c

```diff
@@ -465,6 +465,51 @@ void AppLayerHtpNeedFileInspection(void)
     SCReturn;
 }
 
+static void AppLayerHtpSetStreamDepthFlag(void *tx, uint8_t flags)
+{
+    HtpTxUserData *tx_ud = (HtpTxUserData *) htp_tx_get_user_data((htp_tx_t *)tx);
+    if (tx_ud) {
+        if (flags & STREAM_TOCLIENT) {
+            tx_ud->tcflags |= HTP_STREAM_DEPTH_SET;
+        } else {
+            tx_ud->tsflags |= HTP_STREAM_DEPTH_SET;
+        }
+    }
+}
+
+static int AppLayerHtpCheckStreamDepth(const HtpTxUserData *tx_ud)
+{
+    if (tx_ud->tcflags & HTP_STREAM_DEPTH_SET) {
+        if (tx_ud->response_body.content_len_so_far < FileReassemblyDepth() ||
+            FileReassemblyDepth() == 0) {
+            return 1;
+        }
+    } else if (tx_ud->tsflags & HTP_STREAM_DEPTH_SET) {
+        if (tx_ud->request_body.content_len_so_far < FileReassemblyDepth() ||
+            FileReassemblyDepth() == 0) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+static void AppLayerHtpComputeChunkLength(uint64_t content_len_so_far, uint32_t body_limit,
+                                          uint32_t stream_depth, uint8_t flags, uint32_t *len)
+{
+    if (!(flags & HTP_STREAM_DEPTH_SET) && body_limit > 0 &&
+        (content_len_so_far < body_limit) &&
+        (content_len_so_far + *len) > body_limit)
```

Comment:
I'm a bit worried with the mixing of u64 and u32 here.

Reply:
casting `uint32_t` vars to `uint64_t` should be safer